src/const_eval.md

```diff
@@ -6,13 +6,13 @@ can be evaluated at compile-time.
 
 ## Constant expressions
 
-Certain types of expressions can be evaluated at compile time. These are called
-_constant expressions_ and are required in const contexts. In
-other places, such as in [`let` statements](statements.html#let-statements),
-constant expressions may be evaluated at compile time. If errors, such as out
-of bounds [array indexing] or [overflow] occurs,
-then it is a compiler error if the value must be evaluated at compile time,
-otherwise it is just a warning, but the code will most likely panic when run.
+Certain types of expressions, called constant expressions, can be evaluated at
+compile time. In [const contexts](#const-context), these are the only allowed
+expressions, and are always evaluated at compile time. In other places, such as
+in [`let` statements](statements.html#let-statements), constant expressions are
+Errors such as out of bounds [array indexing] or [overflow] are compiler errors
```

Comment:
I think:
> allowed but not required, and may be evaluated at compile time.

accidentally fell off?

Reply:
Also, given that constant expressions are a strict subset of all expressions, is "allowed but not required" not implied?

Reply:
Ah, good catch on the missing bit; I'll push a change with that and also clarifying the other language.

Reply:
And, there we go.